src/Azure.Functions.Cli/StaticResources/Dockerfile.node18

```diff
@@ -5,7 +5,12 @@ FROM mcr.microsoft.com/azure-functions/node:4-node18
 ENV AzureWebJobsScriptRoot=/home/site/wwwroot \
     AzureFunctionsJobHost__Logging__Console__IsEnabled=true
 
-COPY . /home/site/wwwroot
+WORKDIR /home/site/wwwroot
 
-RUN cd /home/site/wwwroot && \
-    npm install
\ No newline at end of file
+# Copy package.json and install packages before copying the rest of the code to enable caching
+COPY package.json package.json
+RUN npm install
+
+# Copy the rest of the code
+COPY . .
+RUN  npm run build
```

Comment:
Hi @stuartleeks sorry for the delay and thanks for the contribution! Please remove the `npm run build` step from the 3 files without `.typescript` in the name. These 3 files should be for JavaScript, which doesn't need a build step and didn't have one before your change

Reply:
Oops - good catch, thanks!

Reply:
Updated and rebased on latest `v4.x`